src/jit/lclvars.cpp

```diff
@@ -1748,18 +1748,38 @@ void Compiler::lvaPromoteStructVar(unsigned lclNum, lvaStructPromotionInfo* Stru
 //
 void Compiler::lvaPromoteLongVars()
 {
-    if ((opts.compFlags & CLFLG_REGVAR) == 0)
+#ifdef DEBUG
+    if (verbose)
+    {
+        printf("*************** In lvaPromoteLongVars()\n");
+    }
+#endif
+
+    if (!opts.OptEnabled(CLFLG_REGVAR))
     {
+        JITDUMP("CLFLG_REGVAR not set; not promoting long vars\n");
         return;
     }
+
     // The lvaTable might grow as we grab temps. Make a local copy here.
     unsigned startLvaCount = lvaCount;
     for (unsigned lclNum = 0; lclNum < startLvaCount; lclNum++)
     {
         LclVarDsc* varDsc = &lvaTable[lclNum];
-        if (!varTypeIsLong(varDsc) || varDsc->lvDoNotEnregister || varDsc->lvIsMultiRegArgOrRet() ||
-            (varDsc->lvRefCnt == 0))
+        if (!varTypeIsLong(varDsc))
+        {
+            continue;
+        }
+
+        if (varDsc->lvIsMultiRegArgOrRet())
+        {
+            JITDUMP("Not promoting long local V%02u: marked lvIsMultiRegArgOrRet()\n", lclNum);
+            continue;
+        }
+
+        if (varDsc->lvRefCnt == 0)
         {
+            JITDUMP("Not promoting long local V%02u: lvRefCnt == 0\n", lclNum);
```

Comment:
It would be useful to have such an assert, I would think (I could have sworn there was one, and perhaps there is, but I couldn't find it). There's such an assert in `emitter:emitEndCodeGen`, but it only applies to GC refs.